uint8[] mode
float64[] reference
float64[] manual
float64[] offset

